File: StackExchange Client/src/java/controllers/CommentController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package controllers;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.PrintWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

/**
 *
 * @author Tifani
 */
public class CommentController extends HttpServlet {
    // Path for the Identity Service
    private final static String CONTEXT_PATH = "http://localhost:8083/StackExchange_VoteComment/";

    /**
     * Handles the HTTP <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String q_id = request.getParameter("q_id");
        try (PrintWriter out = response.getWriter()){
            String query = String.format("?q_id=%s", q_id);
            URL url = new URL(CONTEXT_PATH + "/comment"+query);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("GET");

            // Buffer the result into a string
            BufferedReader rd = new BufferedReader(
                    new InputStreamReader(conn.getInputStream()));
            StringBuilder sb = new StringBuilder();
            String line;
            while ((line = rd.readLine()) != null) {
                sb.append(line);
            }

            rd.close();
            //conn.disconnect();
            
            //Print result
            out.print(sb.toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * Handles the HTTP <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String q_id = request.getParameter("q_id");
        String content = request.getParameter("content");
        String token = request.getParameter("token");
        
        String query = String.format("q_id=%s&content=%s&token=%s",
                q_id, content, token);
        
        try (PrintWriter output = response.getWriter()){
            // Establish HTTP connection with Identity Service
            URL url = new URL(CONTEXT_PATH + "comment");
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("POST");
            conn.setDoOutput(true);
            conn.setDoInput(true);
            conn.setUseCaches(false);
            conn.setAllowUserInteraction(false);
            conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
            
            //Create the form content
            try (OutputStream out = conn.getOutputStream()) {
                out.write(query.getBytes());
                out.close();
            }
            
            // Buffer the result into a string
            BufferedReader rd = new BufferedReader(
                    new InputStreamReader(conn.getInputStream()));
            StringBuilder sb = new StringBuilder();
            String line;
            while ((line = rd.readLine()) != null) {
                sb.append(line);
            }
            
            rd.close();
            //conn.disconnect();
            
            //Print result
            output.print(sb.toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
  
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}

File: StackExchange_Client/web/js/controllers/commentController.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
app.controller('commentListController', ['$scope','$http','$location',
  function($scope,$http,$location) {
  var qid = $location.search()['id'];
    var req = {
   method: "GET",
   url: "http://localhost:8083/Comment-Vote_Service/comments",
   params: {id: qid}
  }
    $http(req)
    .then(function(response) {$scope.comments = response.data.comments;});
}]); 

app.controller('addCommentController', ['$scope','$http','$location','$cookies',
  function($scope,$http,$location,$cookies) {
    $scope.submitComment = function() {
    var qid = $location.search()['id'];
    var token = $cookies.get("stackexchange_token");
    var content = $scope.newcontent;
    var submitreq = {
      method: "POST",
      url: "http://localhost:8083/Comment-Vote_Service/addComment",
      headers: {'Content-Type': 'application/x-www-form-urlencoded'},
      transformRequest: function(obj) {
        var str = [];
        for(var p in obj)
        str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
        return str.join("&");
      },
      data: {id: qid, content: content, token: token}
    }
    $http(submitreq)
    .then(function(response) {alert("tes");});
  }
}]); 



